olive/cli/finetune.py

```diff
@@ -0,0 +1,267 @@
+        dataset_group.add_argument(
+            "--data_files", type=str, help="The dataset files. If multiple files, separate by comma."
+        )
+        dataset_group.add_argument("--train_split", type=str, default="train", help="The split to use for training.")
+        dataset_group.add_argument(
+            "--eval_split",
+            default="",
+            help="The dataset split to evaluate on.",
+        )
+        text_group = dataset_group.add_mutually_exclusive_group(required=True)
+        text_group.add_argument(
+            "--text_field",
+            type=str,
+            help="The text field to use for fine-tuning.",
+        )
+        text_group.add_argument(
+            "--text_template",
+            # using special string type to allow for escaped characters like \n
+            type=unescaped_str,
+            help=r"Template to generate text field from. E.g. '### Question: {prompt} \n### Answer: {response}'",
+        )
+        dataset_group.add_argument(
+            "--max_seq_len",
+            type=int,
+            default=1024,
+            help="Maximum sequence length for the data.",
+        )
+        # lora options
+        lora_group = sub_parser.add_argument_group("lora options")
+        lora_group.add_argument(
+            "--method",
+            type=str,
+            default="lora",
+            choices=["lora", "qlora"],
+            help="The method to use for fine-tuning",
+        )
+        lora_group.add_argument(
+            "--lora_r",
+            type=int,
+            default=64,
+            help="LoRA R value.",
+        )
+        lora_group.add_argument(
+            "--lora_alpha",
+            type=int,
+            default=16,
+            help="LoRA alpha value.",
+        )
+        # peft doesn't know about phi3, should we set it ourself in the lora pass based on model type?
+        lora_group.add_argument(
+            "--target_modules", type=str, help="The target modules for LoRA. If multiple, separate by comma."
+        )
+
+        # directory options
+        sub_parser.add_argument("-o", "--output_path", type=str, default="optimized-model", help="Output path")
+        sub_parser.add_argument(
+            "--tempdir", default=None, type=str, help="Root directory for tempfile directories and files"
+        )
+        # TODO(jambayk): what about checkpoint_dir and resume from checkpoint support? clean checkpoint dir?
+        sub_parser.add_argument("--clean", action="store_true", help="Run in a clean cache directory")
+
+        sub_parser.set_defaults(func=FineTuneCommand)
+
+    def run(self):
+        from olive.workflows import run as olive_run
+
+        set_tempdir(self.args.tempdir)
+
+        run_config = self.get_run_config()
+        with tempfile.TemporaryDirectory() as tempdir:
+            run_config["output_dir"] = tempdir
+            olive_run(run_config)
+
+            # need to improve the output structure of olive run
```

Comment:
Agree with this.
Olive workflow will copy same model multi-times. Is it possible only run fine-tune passes out of engine?  I met the similar concern in perf-tuning cli.

Reply:
We definitely need to reconsider the output structure. Before when the engine was run in no search mode, its output was a dict with model config but now it gives us a dictionary of accelerator:footprint which is not easy to parse 

We have similar issues in the other example scripts too. Where we have to hardcode the expected output structure.

Reply:
Let us add it into todo items.